src/include/txn_inline.h

```diff
@@ -363,34 +363,54 @@ __wt_txn_op_delete_apply_prepare_state(WT_SESSION_IMPL *session, WT_REF *ref, bo
  * __txn_op_delete_commit_apply_page_del_timestamp --
  *     Apply the correct start and durable timestamps to the page delete structure.
  */
-static WT_INLINE void
-__txn_op_delete_commit_apply_page_del_timestamp(WT_SESSION_IMPL *session, WT_REF *ref)
+static WT_INLINE int
+__txn_op_delete_commit_apply_page_del_timestamp(WT_SESSION_IMPL *session, WT_TXN_OP *op)
 {
+    WT_ADDR_COPY addr;
+    WT_DECL_RET;
     WT_PAGE_DELETED *page_del;
+    WT_REF *ref;
     WT_TXN *txn;
+    bool addr_found;
 
+    ref = op->u.ref;
     txn = session->txn;
     page_del = ref->page_del;
 
     WT_ASSERT(session, WT_REF_GET_STATE(ref) == WT_REF_LOCKED);
 
     if (page_del != NULL && page_del->timestamp == WT_TS_NONE) {
+
+        /* Validate the commit timestamp against the maximum durable timestamp on the page. */
+        WT_ENTER_GENERATION(session, WT_GEN_SPLIT);
+        WT_WITH_BTREE(session, op->btree, addr_found = __wt_ref_addr_copy(session, ref, &addr));
+        if (addr_found)
```

Comment:
Shouldn't this be an assert, as a deleted page should always have an addr?

Reply:
Sometimes a deleted page may not have an addr. But in this scenario, as it is a fast truncated page, it must have an address. I changed the `if` check into an assert_always.